src/adapter/src/client.rs

```diff
@@ -172,22 +173,55 @@ impl Client {
         };
         let response = client
             .send(|tx, session| Command::Startup {
-                session,
                 cancel_tx,
                 tx,
                 set_setting_keys,
+                user: session.user().clone(),
+                conn_id: session.conn_id().clone(),
```

Comment:
Assuming I understand where the split will take place, one day we'll need to serialize `conn_id` and send it across the wire, right? I think this should be fine but I just want to call out that `ConnectionId` is an owned, ref-counted type, where we only release the ID when all handles are dropped.

But it looks like we keep the `Session` alive so this should be fine?

Reply:
Yes, one day we will need to make conn id network safe. There's other things too (the notice channel for example), but session is a great start.